### PR TITLE
IsSession on auth token and validate admin creds on setup

### LINKED ISSUE #99: Treat browser tokens and service tokens separately
1. Browser tokens obtained via login should not appear on the API keys page. They should not be revokable through that UI - and should instead be revoked only via sign out
2. You should not be able to authenticate with an API key. Instead you must login with username/password to get a browser token, which will be revoked on signout. Browser tokens should not be treated like service tokens, which are created via the API keys page, and vice versa

- this prevents you from revoking your session's token via the API key page
- it also prevents you from setting up a token for a service that is also used for your browser session, and then signing out and causing that service token from being unintentionally revoked

cc @jimmylee

### PR DESCRIPTION
**Pull request by neelvirdy**
Along with https://github.com/application-research/estuary-www/pull/164, fixes https://github.com/application-research/estuary-www/issues/99 and https://github.com/application-research/estuary/issues/815

1. Adds `IsSession` field on AuthToken table and populates with `false` for tokens created via api admin page, and `true` for logins/signups. Session tokens will not be rendered in the api admin page so users cannot mix up their service and session tokens. The frontend will also no longer support authenticating via API key to prevent using a service token as a session token. That feature is also unnecessary when admin login is fixed.
2. When running `./estuary setup`, validates admin username and password to be compatible with bcrypt's requirement of 1-32 alphanumeric characters on usernames and 8+ alphanumeric characters on passwords.

image
image

I got a success with a valid creds, but omitted it from the screenshot.

### CODE REVIEW

**Comment by en0ma** [comment]
@neelvirdy how did you come up with the term `session` token for the admin API key?

**Comment by en0ma** [comment]
> Session tokens will not be rendered in the api admin page so users cannot mix up their service and session tokens

Makes sense, since it's a one-time setup key (session?) no need to show it in the UI. My question now is, does this mean the admin after logging in will have to then create a service API key if they need a key?

**Comment by en0ma** [comment]
> When running ./estuary setup, validates admin username and password to be compatible with bcrypt's requirement of 1-32 alphanumeric characters on usernames and 8+ alphanumeric characters on passwords.

Would this play nicely with already existing credentials?

**Comment by neelvirdy** [comment]
> @neelvirdy how did you come up with the term `session` token for the admin API key?

The session field actually doesn't correspond to admin vs regular user. It's purely whether or not the token was generated through login/signup, or if it was a manually provisioned key. In otherwords, session tokens will live in browser cookies exclusively. Service tokens are for all other use cases exclusively (scripts, curls, and other services calling estuary).

The admin creds are relevant to this change because admin login on estuary-www was not working, which forced us to authenticate via API key instead. But, authenticating via API key introduces a risk of mixing up session vs service tokens which defeats some of the purpose of the distinction. This is why the linked estuary-www PR fixes the admin login in addition to making the change to not render session tokens.




> > Session tokens will not be rendered in the api admin page so users cannot mix up their service and session tokens
> 
> Makes sense, since it's a one-time setup key (session?) no need to show it in the UI. My question now is, does this mean the admin after logging in will have to then create a service API key if they need a key?

Login creates a key, so the admin will get their first key simply by logging in with the credentials they ran `./estuary setup` with.

> > When running ./estuary setup, validates admin username and password to be compatible with bcrypt's requirement of 1-32 alphanumeric characters on usernames and 8+ alphanumeric characters on passwords.
> 
> Would this play nicely with already existing credentials?

Bcrypt already throws an error if these requirements are not met, we are simply catching it earlier so the user can't create a user in DB with invalid credentials, only to get a login failure down the road.

File: main.go [hunk 1]
```diff
@@ -565,20 +575,6 @@ func main() {
 				if err := db.Create(newUser).Error; err != nil {
 					return fmt.Errorf("admin user creation failed: %w", err)
 				}
-
```
Comment (en0ma) [inline main.go]: please update the README that running `estuary setup` will no longer return an API key

Comment (en0ma) [reply main.go]: @neelvirdy after the readme is updated, I think this is ready to go

Comment (neelvirdy) [reply main.go]: updated

**Review by en0ma — DISMISSED** [review]
LGTM

File: .github/workflows/postman-tests.yml [hunk 2]
```diff
@@ -41,7 +41,7 @@ jobs:
         npm install
 
     - name: Setup
-      run: echo "APIKEY=$(./estuary setup --username admin --password password  |  sed -e's/.*EST/EST/g' | tail -n1)" >> $GITHUB_ENV
+      run: echo "APIKEY=$(./estuary setup --username admin --password password1  |  sed -e's/.*EST/EST/g' | tail -n1)" >> $GITHUB_ENV
```
Comment (en0ma) [inline .github/workflows/postman-tests.yml:44]: @neelvirdy why this change?

**Review by en0ma — APPROVED** [review]
LGTM

**Review by alvin-reyes — APPROVED** [review]
LGTM

File: .github/workflows/postman-tests.yml (hunk 2 above)
Comment (neelvirdy) [reply .github/workflows/postman-tests.yml:44]: password is an invalid password which would now fail the setup cmd
